pkg/operator/operator.go

```diff
@@ -169,7 +170,7 @@ func getCABundle(restConfig *rest.Config) (*string, error) {
 }
 
 func getVnetGUID(cfg *auth.Config, subnetID string) (string, error) {
-	creds, err := auth.NewCredential(cfg)
+	creds, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
An alternative is to keep the abstraction, but just return `azidentity.NewDefaultAzureCredential(nil)`. But there are only 2 uses (another in azure_client.go) at the moment, so I don't think it is worth it.